compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -801,6 +801,15 @@ impl GenericParamDefKind {
             GenericParamDefKind::Const => "constant",
         }
     }
+    pub fn to_ord(&self, sess: &Session) -> ast::ParamKindOrd {
+        match self {
+            GenericParamDefKind::Lifetime => ast::ParamKindOrd::Lifetime,
+            GenericParamDefKind::Type { .. } => ast::ParamKindOrd::Type,
+            GenericParamDefKind::Const => {
+                ast::ParamKindOrd::Const { unordered: sess.features_untracked().const_generics }
```

Comment:
```suggestion
                ast::ParamKindOrd::Const { unordered: tcx.features().const_generics }
```
`features_untracked()` can result in incorrect diagnostics if a query doesn't get invalidated by a changing feature.
If used outside of diagnostic improvements it could even be unsound.

So even if it's probably fine to use `features_untracked` at the only place these methods are currently used, please change this to use the `tcx.features()` which is tracked by the query system.

Reply:
Should I change the HIR as well? I can't directly add it as it would introduce a circular dep, but I could pass features instead.